lib/sgid-client.service.js

```diff
@@ -0,0 +1,63 @@
+const { SgidClient } = require('@opengovsg/sgid-client')
+
+class SgidService {
+  DEFAULT_NONCE = 'randomnonce'
+
+  constructor(baseUrl, clientId, clientSecret, redirectUri, privateKey) {
+    this.sgidClient = new SgidClient({
+      clientId: clientId,
+      clientSecret: clientSecret,
+      privateKey: privateKey,
+      redirectUri: redirectUri,
+      hostname: baseUrl,
+    })
+    this.redirectUri = redirectUri
+  }
+
+  /**
+   * Fetches the token via sgid SDK
+   *
+   * @param {string} code
+   * @returns {object} { sub: string, accessToken: string }
+   */
+  async fetchToken(code) {
+    if (!code) throw Error(`code cannot be empty`)
+
+    try {
+      const { sub, accessToken } = await this.sgidClient.callback(
+        code,
+        this.DEFAULT_NONCE,
+        this.redirectUri,
+      )
+      return { sub, accessToken: accessToken }
+    } catch (e) {
+      console.error(e)
+      throw new Error(
+        'Error retrieving access token via sgid-client'
+      )
+    }
+  }
+
+  /**
+   * Fetches the user info via sgid SDK
+   *
+   * @param {string} baseUrl
+   * @param {string} accessToken
+   * @return {object} { sub: string, data: array }
+   */
+  async fetchUserInfo(baseUrl, accessToken) {
+    for (const [key, value] of Object.entries({ baseUrl, accessToken })) {
+      if (!value) throw Error(`${key} cannot be empty`)
+    }
```

Comment:
nit: i know that this was from the previous version of the code, but i think it's a bit confusing to read. might be clearer if we just do something like `if (!baseUrl) throw Error...` for both variables. what do you think?

Reply:
I agree actually, since I think we won't be appending this function in the near future we don't have to worry about extensability.